ui/ozone/platform/wayland/wayland_window.h

```diff
@@ -39,6 +39,16 @@ class WaylandWindow : public PlatformWindow, public PlatformEventDispatcher {
   // Set whether this window has keyboard focus and should dispatch key events.
   void set_keyboard_focus(bool focus) { has_keyboard_focus_ = focus; }
 
+  // Set a child of this window. It is very important in case of nested
+  // xdg_popups as long as we must destroy the very last first and only then
+  // its parent.
+  void set_child_window(WaylandWindow* window) { child_window_ = window; }
+
+  bool has_focus() { return has_pointer_focus_ || has_keyboard_focus_; }
```

Comment:
has_keyboard_or_pointer_focus? and move to line 42, right below the other focus methods?